src/storage/patch/ImmutableMetadataPatcher.ts

```diff
@@ -0,0 +1,78 @@
+ * List of triples that must not be updated are given during instantiation.
+ *
+ * This list is a list of tuples where the first element is the predicate and the second element the object.
+ * Thus, when a triple has been added/removed in the metadata resource
+ * where its predicate is equal to the first element and its object is equal to the second element,
+ * then this Patcher throws a ConflictHttpError as this is not allowed.
+ * Note that the object can be an empty string,
+ * which will be interpreted that the object of the modified triple can be anything.
+ */
+export class ImmutableMetadataPatcher extends RdfStorePatcher {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly patcher: RdfStorePatcher;
+  private readonly metadataStrategy: AuxiliaryStrategy;
+  private readonly immutableTriples: [string, string][];
+
+  public constructor(patcher: RdfStorePatcher, metadataStrategy: AuxiliaryStrategy,
+    immutableTriples: [string, string][]) {
+    super();
+    this.patcher = patcher;
+    this.metadataStrategy = metadataStrategy;
+    this.immutableTriples = immutableTriples;
+  }
+
+  public async canHandle({ identifier }: RdfStorePatcherInput): Promise<void> {
+    if (!this.metadataStrategy.isAuxiliaryIdentifier(identifier)) {
+      throw new NotImplementedHttpError('Only metadata resources can be checked on immutable triples.');
+    }
+  }
+
+  public async handle({ store, identifier, patch }: RdfStorePatcherInput): Promise<Store> {
+    const inputStore = store;
+    const patchedStore = await this.patcher.handleSafe({
+      identifier,
+      patch,
+      store: new Store(store.getQuads(null, null, null, null)),
+    });
+
+    const inputImmutable: Quad[] = [];
+    const patchedImmutable: Quad[] = [];
+
+    // For loop over triples that can not be changed in solid metadata
+    for (const immutable of this.immutableTriples) {
+      const predicate = immutable[0];
+      const object = immutable[1] !== '' ? immutable[1] : null;
+      inputImmutable.push(...inputStore.getQuads(null, predicate, object, null));
+      patchedImmutable.push(...patchedStore.getQuads(null, predicate, object, null));
+    }
+
+    this.logger.debug(`input stream immutable: ${await readableToString(serializeQuads(inputImmutable))}`);
+    this.logger.debug(`output stream immutable: ${await readableToString(serializeQuads(patchedImmutable))}`);
+
+    // Filter out differences using custom filter
+    const unique = uniqueQuads([ ...inputImmutable, ...patchedImmutable ]);
+    const lengthTogether = unique.length;
+
+    this.logger.debug(`Nothing has changed: ${lengthTogether === inputImmutable.length &&
+        lengthTogether === patchedImmutable.length}`);
+    if (!(lengthTogether === inputImmutable.length && lengthTogether === patchedImmutable.length)) {
+      throw new ConflictHttpError('Not allowed to change this type of metadata.');
+    }
```

Comment:
The new changed check did not behave as intended when testing locally for one case using the config `file-no-setup.json `:
```bash
curl --location --request PATCH 'http://localhost:3123/.meta' --header 'Content-Type: application/sparql-update' --data-raw 'PREFIX dc: <http://purl.org/dc/elements/1.1/>
INSERT DATA
{ 
    <http://localhost:3123/#Collection> a <https://w3id.org/ldes#EventStream>.
}'
{"name":"ConflictHttpError","message":"Not allowed to change this type of metadata.","statusCode":409,"errorCode":"H409","stack":"ConflictHttpError: Not allowed to change this type of metadata.\n    at ImmutableMetadataPatcher.handle (/home/wouts/Documents/metadataEditing/community-server/dist/storage/patch/ImmutableMetadataPatcher.js:61:19)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async RdfPatcher.handle (/home/wouts/Documents/metadataEditing/community-server/dist/storage/patch/RdfPatcher.js:30:29)\n    at async ConvertingPatcher.handle (/home/wouts/Documents/metadataEditing/community-server/dist/storage/patch/ConvertingPatcher.js:58:22)\n    at async RepresentationPatchHandler.handle (/home/wouts/Documents/metadataEditing/community-server/dist/storage/patch/RepresentationPatchHandler.js:39:25)\n    at async runWithTimeout (/home/wouts/Documents/metadataEditing/community-server/dist/util/locking/WrappedExpiringReadWriteLocker.js:51:24)\n    at async GreedyReadWriteLocker.withWriteLock (/home/wouts/Documents/metadataEditing/community-server/dist/util/locking/GreedyReadWriteLocker.js:48:20)\n    at async MonitoringStore.modifyResource (/home/wouts/Documents/metadataEditing/community-server/dist/storage/MonitoringStore.js:32:33)\n    at async PatchOperationHandler.handle (/home/wouts/Documents/metadataEditing/community-server/dist/http/ldp/PatchOperationHandler.js:39:9)"}
```
Thus this suggested change is not used.

Reply:
It's not because there's an error in my example code that we shouldn't look into having a more efficient solution. 😉 

```ts
const changed = inputImmutable.some((inputQuad): boolean => !patchedImmutable
      .some((patchedQuad): boolean => inputQuad.equals(patchedQuad));
```

That should fix it. I put the negation operator in the wrong place. Unless there's something else that I missed.